docs/naps/template.md

```diff
@@ -0,0 +1,91 @@
+---
+Author: \<list of authors' real names and, optionally, email addresses>
+Created: \<date created on, in yyyy-mm-dd format>
+Resolution: \<url> (required for Accepted | Rejected | Withdrawn)
+Resolved: \<date resolved, in yyyy-mm-dd format>
+Status: |-
+  \<Draft | Active | Accepted | Deferred | Rejected | Withdrawn |
+  Final | Superseded>
+Type: \<Standards Track | Process>
+Version effective: \<version-number> (for accepted NAPs)
+---
+
+(nap-template)=
+
+# NAP-X — Template and Instructions
+
+## Abstract
+
+The abstract should be a short description of what the NAP will achieve.
+
+## Motivation and Scope
+
+This section describes the need for the proposed change. It should describe
+the existing problem, who it affects, what it is trying to solve, and why.
+This section should explicitly address the scope of and key requirements
+for the proposed change.
+
+## Detailed Description
+
+This section should provide a detailed description of the proposed change. It
+should include examples of how the new functionality would be used, intended
+use-cases, and pseudocode illustrating its use.
+
+## Related Work
+
+This section should list relevant and/or similar technologies, possibly in
+other libraries. It does not need to be comprehensive, just list the major
+examples of prior and relevant art.
+
+## Implementation
+
+This section lists the major steps required to implement the NAP. Where
+possible, it should be noted where one step is dependent on another, and which
+steps may be optionally omitted. Where it makes sense, each step should
+include a link related pull requests as the implementation progresses.
+
+Any pull requests or development branches containing work on this NAP
+should be linked to from here. (A NAP does not need to be implemented in a
+single pull request if it makes sense to implement it in discrete phases).
+
+## Backward Compatibility
+
+This section describes the ways in which the NAP breaks backward
+compatibility.
```

Comment:
Should this section not also elaborate on any design decisions made to support backward compatibility?